### PR TITLE
Reader Caterpillars: hide all

### PR DESCRIPTION
**Pull request by samouri**
Implements `hide all` as described by: https://github.com/Automattic/wp-calypso/issues/17663#event-1241222844

### CODE REVIEW

**Comment by samouri** [comment]
assumption I made: we only want hide-all for the top-most caterpillar. is this right?

File: client/blocks/conversation-caterpillar/index.jsx [hunk 1]
```diff
@@ -96,6 +104,13 @@ class ConversationCaterpillarComponent extends React.Component {
 		const lastAuthorName = get( last( displayedAuthors ), 'name' );
 		const gravatarSmallScreenThreshold = MAX_GRAVATARS_TO_DISPLAY / 2;
 
+		const hideShowAllHandler = isEmpty( hiddenComments )
+			? this.hideAllComments
+			: this.handleShowAll;
+		const hideShowAllText = isEmpty( hiddenComments )
+			? translate( 'Hide all' )
```
Comment (a8ci18n) [inline client/blocks/conversation-caterpillar/index.jsx]: Hi! I've found a possible matching string that has already been translated **4** times:
`translate( 'Hide All' )` *ES Score: 11*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by samouri** [comment]
Note: this currently doesn't hide the initial set.  will need to move things around to handle that and don't want to make it conflict with client side hwm.

@blowery: do you think it makes sense to fully go off redux and ignore the postkey comments?

**Comment by blowery** [comment]
@samouri maybe. we'll still need the comments on the postKeys in flux to make update acceptance work, but we could mirror that info out to redux? Have to think that through. We just need to be able to split off "what we have" from "what we're showing".

**Comment by samouri** [comment]
@blowery: we already have that separation with the [expansions](https://github.com/Automattic/wp-calypso/pull/17965/files#diff-512900ef67147a63a42ecc06c3abcdbbR192) reducer.

During an update acceptance event we could run a `reduxDispatch( expandComments( postKey.comments ) )

**Comment by jancavan** [comment]
> we only want hide-all for the top-most caterpillar. is this right?

This is correct.

File: client/blocks/conversation-caterpillar/index.jsx [hunk 2]
```diff
@@ -96,6 +104,11 @@ class ConversationCaterpillarComponent extends React.Component {
 		const lastAuthorName = get( last( displayedAuthors ), 'name' );
 		const gravatarSmallScreenThreshold = MAX_GRAVATARS_TO_DISPLAY / 2;
 
+		const hideShowAllHandler = isEmpty( hiddenComments ) ? this.handleHideAll : this.handleShowAll;
+		const hideShowAllText = isEmpty( hiddenComments )
+			? translate( 'Hide all' )
```
Comment (a8ci18n) [inline client/blocks/conversation-caterpillar/index.jsx]: Hi! I've found a possible matching string that has already been translated **4** times:
`translate( 'Hide All' )` *ES Score: 11*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by jancavan** [comment]
We've removed "Show all" in: https://github.com/Automattic/wp-calypso/pull/18766 so closing this for now.

**Pull request closed without merging**